pkg/syncer/storage/storage_pv_process.go

```diff
@@ -0,0 +1,109 @@
+/*
+Copyright 2022 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package storage
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	kcpcache "github.com/kcp-dev/apimachinery/pkg/cache"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/klog/v2"
+
+	workloadv1alpha1 "github.com/kcp-dev/kcp/pkg/apis/workload/v1alpha1"
+	"github.com/kcp-dev/kcp/pkg/logging"
+	"github.com/kcp-dev/kcp/pkg/syncer/shared"
+)
+
+func (c *PersistentVolumeController) process(ctx context.Context, key string) error {
+	logger := klog.FromContext(ctx)
+
+	clusterName, _, name, err := kcpcache.SplitMetaClusterNamespaceKey(key)
+	if err != nil {
+		return err
+	}
+
+	logger = logger.WithValues(logging.WorkspaceKey, clusterName, logging.NameKey, name)
+
+	upstreamPVObject, err := c.getUpstreamPersistentVolume(clusterName, name)
+	if err != nil {
+		return fmt.Errorf("failed to get upstream PersistentVolume: %w", err)
+	}
+
+	upstreamPV, ok := upstreamPVObject.(*corev1.PersistentVolume)
+	if !ok {
+		return fmt.Errorf("failed to assert object to a PersistentVolume: %T", upstreamPVObject)
+	}
+
+	resourceState, ok := upstreamPV.GetLabels()[workloadv1alpha1.ClusterResourceStateLabelPrefix+c.syncTarget.key]
+	if ok && resourceState == string(workloadv1alpha1.ResourceStateUpsync) {
+		// Fetch the namespace locator annotation from the downstream PVC, this will be helpful to
+		// get the upstream PVC
+		nsLocatorJSON, ok := upstreamPV.GetAnnotations()[shared.NamespaceLocatorAnnotation]
```

Comment:
The namespace locator annotation is on downstream resources, to indicate the corresponding upstream information.

Reply:
So here you would just have to create a namespace locator containing the same information as what you downstream PV would contain. You should have all the information (upstream workspace, namespace, and synctarget info)

Reply:
Like this (found in SpecProcess):
```
	desiredNSLocator := shared.NewNamespaceLocator(clusterName, c.syncTargetWorkspace, c.syncTargetUID, c.syncTargetName, upstreamNamespace)
	jsonNSLocator, err := json.Marshal(desiredNSLocator)
	if err != nil {
		return err
	}

	var downstreamNamespace string
       downstreamNamespaces, err := c.downstreamNSInformer.Informer().GetIndexer().ByIndex(byNamespaceLocatorIndexName, string(jsonNSLocator))
```

Reply:
The PVC upstream namespace you would find in the upstream PV claimref.